src/pipe_mgr/shared/dal/dpdk/dal_init.c

```diff
@@ -114,6 +114,7 @@ int dal_enable_pipeline(bf_dev_id_t dev_id,
 		snprintf(buffer, sizeof(buffer), "gcc -c -O3 -fpic "
```

Comment:
I think the -1 return value could be caused by race condition of snprintf.
running the original code with gdb will get return value of 0.

Reply:
Ahh, that can explain why this occurs probabilistically! Do you know how to avoid the race condition here?

Reply:
track this issue separately